File: Python3/프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    if n % 2 != 0:
        if n == 1:
            return 1
        return n + solution(n-2)
    
    if n % 2 == 0:
        if n == 2:
            return 2**2
        return n**2 + solution(n-2)

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        if (n % 2 != 0){
            for (int i = n; i > 0; i-=2){
                answer += i;
            }
        } else {
            for (int i = n; i > 0; i-=2){
                answer += Math.pow(i, 2);
            }
        }
        
        return answer;
    }
}